components/Terminal/commands/filesystem.ts

```diff
@@ -28,7 +28,7 @@ const fileTree: Entry = {
     {
       name: "home",
       children: (state: State) => {
-        const iAm = whoami(state, []);
+        const iAm = whoami(state, []) ?? 'anonymous';
```

Comment:
why are we doing this absolutely everywhere and not just in the `whoami` function?

Reply:
`whoami` is a command so returns `string | undefined`. If you have a cleaner solution, feel free to suggest.

Reply:
where is the harm in making `whoami` just return a string?
in its current implementation we are already setting the return type to just `string`
```ts
export const whoami: CommandHandler = (state, _params): string => {
  return state.session.username ?? 'anonymous';
};
```

Reply:
The problem is the `CommandHandler` type, which redefines the return type to be `string | undefined`. I guess we could make it so that `whoami` isn't of type `CommandHandler` as it should still be assignable?

Reply:
but something returning `string` completely satisfies the type `string | undefined` 
so there is no problem....
also, if we're doing something thats stupid just to satisfy typescript, then we're probably doing something wrong